app/serializers/search_and_compare/course_serializer.rb

```diff
@@ -280,11 +280,15 @@ def full_address(address1:, address2:, address3:, address4:, postcode:)
       [address1, address2, address3, address4, postcode].reject(&:blank?).join("\n")
     end
 
+    def campuses_full_address(address1:, address2:, address3:, address4:, postcode:)
+      [address1, address2, address3, address4].reject(&:blank?).join(", ") + (postcode.present? ? ' ' + postcode : '')
```

Comment:
Just to check, do we really want the postcode tacked onto the end without a comma?

Reply:
```c#
        private string MapAddress(Site school)
        {
            var addressFragments = new List<string>{
                school.Address1,
                school.Address2,
                school.Address3,
                school.Address4
            }.Where(x => !string.IsNullOrWhiteSpace(x));

            var postCode = school.Postcode ?? "";

            return addressFragments.Any()
                ? String.Join(", ", addressFragments) + " " + postCode
                : postCode;
}
```

it then goes thro the scary geocoder

Reply:
we should keep it as in as geocoder works well enough with it as well as it keeps  cost and effort down, the lat/lang is keyed with the address as far as i remember on search and compare.

Maybe another PR once we get rails working without dependencies on its c# counterpart